src/components/Basket.tsx

```diff
@@ -1,12 +1,76 @@
-import React, { Dispatch } from "react";
-
-import styled from "styled-components";
-
-import { Action, State } from "./../reducers/basket";
-import { StoreObject } from "./../types/StoreObject";
-
-import { StoreCtx } from "./App";
-import BasketItem from "./atom/BasketItem";
+import React, { useContext, FC } from 'react';
+import styled from 'styled-components';
+import { StoreObject } from '../types/StoreObject';
+import BasketItem from './atom/BasketItem';
+import purchaseItems from '../artillery/store';
+import {
+    GlobalContext,
+    dispatchPurchaseItems,
+} from '../state/globalState';
+import Button from './atom/Button';
+
+const Basket: FC = () => {
+    const { state, dispatch } = useContext(GlobalContext);
+    const { user, cart } = state;
+    const totalPrice = Object.keys(cart).reduce(
+        (prev: number, so: string) => {
+            const item = (state.inventory.find((e) => e.pk === Number(so))! as StoreObject);
+            return prev + item.price * cart[Number(so)].quantity;
+        },
+        0,
+    );
+
+    const balance = user ? user.balance : 0;
+    const fundsImageUri = balance >= totalPrice ? 'purchase' : 'insufficient';
+    const fundsText = balance >= totalPrice ? 'PURCHASE' : 'INSUFFICIENT';
+    const cartImageUri = Math.max(Object.keys(cart).length) > 0 ? 'food' : 'nofood';
+
+    const userId = user ? user.pk : -1;
+    const dispatchWithdraw = () => dispatchPurchaseItems(dispatch, totalPrice);
+    const purchase = () => purchaseItems(userId, cart)
+        .then((response) => {
+            if (response.ok) {
+                dispatchWithdraw();
+            }
+        });
+
+    const refresh = () => window.location.reload();
```

Comment:
```suggestion
    const refresh = () => window.location.reload();
    
```

Separate concerns visually